### PR TITLE
 pythonPackages.arelle: 2017-08-24 -> 18.3 

### PR DESCRIPTION
**Pull request by Synthetica9**
###### Motivation for this change

Replaces #50269 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Synthetica9** [comment]
cc @roberth

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/arelle/default.nix [hunk 1]
```diff
@@ -16,30 +15,37 @@ buildPythonPackage rec {
...
 
   # arelle-gui is useless without gui dependencies, so delete it when !gui.
   postInstall = lib.optionalString (!gui) ''
     find $out/bin -name "*arelle-gui*" -delete
+  '' +
+  # By default, not the entirety of the src dir is copied. This means we don't
+  # copy the `images` dir, which is needed for the gui version.
+  lib.optionalString (gui) ''
+    targetDir=$out/${python.sitePackages}
+    cp -vr $src/arelle $targetDir
```
Comment (dotlambda) [inline pkgs/development/python-modules/arelle/default.nix]: `-v` is probably not necessary, is it?

Comment (Synthetica9) [reply pkgs/development/python-modules/arelle/default.nix]: No, but it does give extra debug output, which might be useful

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python3.pkgs.arelle

**Comment by dotlambda** [comment]
> Ran 0 tests in 0.000s

Please provide an appropriate `checkPhase`.

**Comment by dotlambda** [comment]
That's strange. Why are the tests only run on x86_64-linux?

**Comment by Synthetica9** [comment]
> > Ran 0 tests in 0.000s
> 
> Please provide an appropriate `checkPhase`.

Tests _are_ actually being ran, but it's not reporting the output. I'll see what I can do

**Comment by Synthetica9** [comment]
> > > Ran 0 tests in 0.000s
> > 
> > 
> > Please provide an appropriate `checkPhase`.
> 
> Tests _are_ actually being ran, but it's not reporting the output. I'll see what I can do

That gave me no end of encoding problems, so the answer seems to be "nothing"

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python3.pkgs.arelle

**Pull request merged**